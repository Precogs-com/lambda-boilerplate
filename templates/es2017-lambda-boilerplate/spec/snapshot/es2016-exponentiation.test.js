const { transform } = require('./utils');

it('expression with primitives', () => {
    const code = transform('2 ** 10');
    expect(code).toMatchSnapshot();
});

it('expression with variables', () => {
    const code = transform('var x = 2; var y = x ** 10;');
    expect(code).toMatchSnapshot();
});

it('exponentiation assignment', () => {
    const code = transform('var x = 2; x **= 10;');
    expect(code).toMatchSnapshot();
});
