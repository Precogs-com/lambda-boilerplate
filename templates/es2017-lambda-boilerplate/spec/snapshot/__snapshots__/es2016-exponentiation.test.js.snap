// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exponentiation assignment 1`] = `
"\\"use strict\\";

var x = 2;x = Math.pow(x, 10)"
`;

exports[`expression with primitives 1`] = `
"\\"use strict\\";

Math.pow(2, 10);"
`;

exports[`expression with variables 1`] = `
"\\"use strict\\";

var x = 2;var y = Math.pow(x, 10);"
`;
